oap-server/server-receiver-plugin/skywalking-register-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/register/provider/handler/v6/grpc/RegisterServiceHandler.java

```diff
@@ -83,34 +86,39 @@ public RegisterServiceHandler(ModuleManager moduleManager) {
         request.getInstancesList().forEach(instance -> {
             ServiceInventory serviceInventory = serviceInventoryCache.get(instance.getServiceId());
 
-            ServiceInstanceInventory.AgentOsInfo agentOsInfo = new ServiceInstanceInventory.AgentOsInfo();
+            JsonObject instanceProperties = new JsonObject();
+            List<String> ipv4s = new ArrayList<>();
+
             for (KeyStringValuePair property : instance.getPropertiesList()) {
                 String key = property.getKey();
                 switch (key) {
-                    case "OSName":
-                        agentOsInfo.setOsName(property.getValue());
+                    case HOST_NAME:
+                        instanceProperties.addProperty(HOST_NAME, property.getValue());
+                    case OS_NAME:
+                        instanceProperties.addProperty(OS_NAME, property.getValue());
                         break;
-                    case "hostname":
-                        agentOsInfo.setHostname(property.getValue());
+                    case LANGUAGE:
+                        instanceProperties.addProperty(LANGUAGE, property.getValue());
                         break;
                     case "ipv4":
```

Comment:
`case "ipv4"` ---> `case IPV4S`

Reply:
**IPV4S** == `ipv4s`, not the same key. `ipv4` is the key used in agent, `OSUtil#buildOSInfo`